### ISSUE TITLE
Bug Report: Entity moves to wrong location in 3D Viewport when resetting scale

### ISSUE THREAD

**Issue by FiniteStateGit** [o3de/o3de#13078]
**Describe the bug**
If an entity is scaled with the scale manipulator while it's pivot point is set to center-of-volume and then the entity scale is reset, the entity will move to the wrong location in the 3D Viewport. Repeatedly scaling and resetting the entity in these conditions will shift the entity in the Z-direction.

While scaling the entity you can see that the Z-translation value in the Transform component is changing.

**Steps to reproduce**
1. Open the default level and select the shaderball.
2. Press **P** to change the shaderball's pivot point to center-of-volume.
3. Press **3** to change to the scale manipulator.
4. Scale the entity by **Left-click + Dragging** the entity.
5. Press **R** to reset the entity's scale.  Observe that the entity has moved in the Z-direction.

You can repeat steps 4-5 for additional observations of this bug. 

**Expected behavior**
Scaling only affects an entity's scale.

**Actual behavior**
Scaling affects both scale and the Z component of it's translation.

**Screenshots/Video**

https://user-images.githubusercontent.com/31811688/200688459-ec2774b6-b92b-4a9a-92df-0f75554287cc.mp4

**Found in Branch**
Development

**Comment by FiniteStateGit** [comment]
@hultonha Correct, the object's world translation does not shift when using the default pivot mode.

**Comment by hultonha** [comment]
> @hultonha Correct, the object's world translation does not shift when using the default pivot mode.

Interesting 🤔 thanks for confirming @FiniteStateGit - not sure when we'll get to this right now but I'll definitely add it to my list and see if I can track down what's going on 👍 Thanks for reporting it.
